bot/resources/tags/mutability.md

```diff
@@ -0,0 +1,32 @@
+**Mutable vs immutable objects**
+
+Imagine that you want to make all letters in a string upper case. Conveniently, strings have an `.upper()` method.
+
+You might think that this would work:
+```python
+string = "abcd"
+string.upper()
+print(string) # abcd
+```
+
+`string` didn't change. Why is that so?
+
+That's because strings in Python are _immutable_. You can't change them, you can only pass around existing strings or create new oness.
+
+```python
+string = "abcd"
+string = string.upper()
+```
+
+`string.upper()` creates a new string which is like the old one, but with allthe letters turned to upper case.
+
+`int`, `float`, `complex`, `tuple`, `frozenset` are other examples of immutable data types in Python.
+
+Mutable data types like `list`, on the other hand, can be changed in-place:
+```python
+my_list = [1, 2, 3]
+my_list.append(4)
+print(my_list) # [1, 2, 3, 4]
+```
+
+`dict` and `set` are other examples of mutable data types in Python.
```

Comment:
```suggestion
Other examples of mutable data types in Python are `dict` and `set`.
```
Just a minor nit-pick, but I don't like that this sentence is a copy of the one on line 23. Reversing it like this would remedy that, but it's not necessary.